packaging.md

```diff
@@ -0,0 +1,156 @@
+# Packaging Applications for Deployment
+
+Once an application is built for production, it still needs to be packaged before it can be deployed to servers. There are several strategies for packaging Swift applications for deployment.
+
+## Docker
+
+One of the most popular ways to package applications these days is using container technologies such as [Docker](https://www.docker.com).
+
+Using Docker's tooling, we can build and package the application as a Docker image, publish it to a Docker repository, and later launch it directly on a server or on a platform that supports Docker deployments such as [Kubernetes](https://kubernetes.io). Many public cloud providers including AWS, GCP, Azure, IBM and others encourage this kind of deployment.
+
+Here is an example `Dockerfile` that builds and packages the application on top of CentoOS:
+
+```Dockerfile
+#------- build -------
+FROM swiftlang/swift:nightly-centos8 as builder
+
+# set up the workspace
+RUN mkdir /workspace
+WORKDIR /workspace
+
+# copy the source to the docker image
+COPY . /workspace
+
+RUN swift build -c release -Xswiftc -g
+
+#------- package -------
+FROM centos:8
+# copy executables
+COPY --from=builder /workspace/.build/release/<executable-name> /
+# copy Swift's dynamic libraries dependencies
+COPY --from=builder /usr/lib/swift/linux/lib*so* /
+
+# set the entry point (application name)
+CMD ["<executable-name>"]
+```
+
+To create a local Docker image from the `Dockerfile` use the `docker build` command from the application's source location, e.g.:
+
+```bash
+$ docker build . -t <my-app>:<my-app-version>
+```
+
+To test the local image use the `docker run` command, .e.g.:
+
+```bash
+$ docker run <my-app>:<my-app-version>
```

Comment:
May I suggest that we amend a bit the style we do code snippets with user filled in things a bit?

It's nice if before we start there's a

```
export MY_APP=<my-app> 
export MY_APP_VERSION=<...>
```

and then in all snippets we use those; this allows people to set their values and copy paste the steps if they want to.

(as silly as it is to encourage copy paste i think it helps some folks and eases the barrier to try things out).

Just an idea